src/profile-logic/merge-compare.js

```diff
@@ -1098,3 +1103,58 @@ function mergeMarkers(
 
   return { markerTable: newMarkerTable, translationMaps };
 }
+
+/**
+ * Merge screenshot markers from different threads. And update the target threads string table while doing it.
+ */
+function mergeScreenshotMarkers(
+  threads: Thread[],
+  targetThread: Thread
+): {
+  markerTable: RawMarkerTable,
+  translationMaps: TranslationMapForMarkers[],
+} {
+  const targetMarkerTable = { ...targetThread.markers };
+  const translationMaps = [];
+
+  threads.forEach(thread => {
+    if (thread.name === 'Compositor') {
+      const translationMap = new Map();
+      const { markers, stringTable } = thread;
+
+      for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+        // We need to move the url string to the new string table if doesn't exist.
+        if (
+          markers.data[markerIndex] !== null &&
+          markers.data[markerIndex].type === 'CompositorScreenshot'
+        ) {
+          const urlIndex = markers.data[markerIndex].url;
+          const newUrl = urlIndex >= 0 ? stringTable.getString(urlIndex) : null;
+
+          // Move compositorScreenshot marker data to the new marker table.
+          const compositorScreenshotMarkerData = {
+            ...markers.data[markerIndex],
```

Comment:
Nice! It's great that you actually didn't mutate the old marker data, and shallow copied to a new object instead!